### DISCUSSION TITLE
toArray() without mapping options ?

### DISCUSSION THREAD

**Discussion by Nir-An** [spatie/laravel-data#183] [Q&A] [answered]
1. There is a way to disable mapping property when using toArray()?

the closest behavior is :
> [Getting a data object without transforming](https://spatie.be/docs/laravel-data/v2/as-a-resource/transformers#content-getting-a-data-object-without-transforming)
It is possible to get an array representation of a data object without transforming the properties. This means Carbon objects won't be transformed into strings. And also, nested data objects and DataCollections won't be transformed into arrays. You can do this by calling the ``all`` method on a data object like this:
```
ArtistData::from($artist)->all();
```

Although  transform has  been disabled, the mapping property is still respected, and actuality  the property names changed

2. If mapping property is a kind of transform, why are property names changed by the ``all()`` method?

**Comment by rubenvanassche** [comment, marked as answer]
Transforming != mapping properties. Transformation is about transforming complex types into more basic types it has nothing to do with mapping property names.

Maybe we cannot update the transform method with a property `mapPropertyNames` which is default `true`, then it can be disabled using the transform method when required. I would accept a PR implementing such feature! The `toArray` method would keep mapping property names by default, but if required you can then use the `transform` method with a flag.

**Comment by Nir-An** [comment]
Hi @rubenvanassche 
I made a [PR for mapPropertyNames](https://github.com/spatie/laravel-data/pull/199)
I implemented it in `transform` function  
but I don't know if it is  right to implement it in[ `all()` function](https://github.com/Nir-An/laravel-data/blob/transform_properties_flexibility/src/Concerns/TransformableData.php#L11) as well, because it should be breaking changes,
nevertheless is simple to remove the flag from `all()` function, and change the tests to  `transform` function.
what do you prefer?
Thanks!

**Reply by rubenvanassche** [reply]
Thanks for your PR @Nir-An, I've added some comments on the PR!
